src/connector/src/source/kafka/source/reader.rs

```diff
@@ -250,7 +242,20 @@ impl KafkaSplitReader {
 
             for (partition, offset) in split_msg_offsets {
                 let split_id = partition.to_string();
-                self.report_latest_message_id(&split_id, offset);
+                latest_message_id_metrics
```

Comment:
move out of the function because it requires a `&mut self` while the loop itself has taken a &self